### PR TITLE
Changes due to workflows.

### PR DESCRIPTION
**Pull request by sinukaarel**
Removed required autoresponder from settings.
Removed api key from settigs.
Cron sync tokens auto generated.
Credentials validation handeled differently when saving settings.
Subdomain parsed form full url.
Workflows api populates autoresponders list.
Formatting and standard changes.
Abandoned cart mininum delay 15min.
Version 1.1.0

### CODE REVIEW

File: opencart 3.0.0.0 - 3.0.3.1/upload/admin/controller/extension/module/smaily_for_opencart.php [hunk 1]
```diff
@@ -392,21 +365,38 @@ public function ajaxValidateCredentials() {
...
+                empty($this->request->post['username']) ||
+                empty($this->request->post['password'])) {
                     // Show empty field error message.
                     $response['error'] = $this->language->get('error_validate_empty');
-                }
+                    echo json_encode($response);
+                    return;
+            }
+
+            $subdomain = $this->db->escape($this->request->post['subdomain']);
```
Comment (kaittodesk) [inline opencart 3.0.0.0 - 3.0.3.1/upload/admin/controller/extension/module/smaily_for_opencart.php]: Why are you escaping the value for database insert before cleaning it?

File: readme.md [hunk 2]
```diff
@@ -1,8 +1,6 @@
 # Smaily for Opencart
 
-> DISCLAIMER Plugin is still in development and currently being reviewed by Opencart. Use at your own discretion before fully finished. All bug reports are welcome under issues tab.
-
-Simple and flexible Smaily integration for Opencart.
+> Simple and flexible Smaily integration for Opencart.
```
Comment (kaittodesk) [inline readme.md]: Remove `>`. No need to make this sentence a blockquote.

File: readme.md [hunk 3]
```diff
@@ -66,6 +64,14 @@ All development for Smaily for Opencart is [handled via GitHub](https://github.c
 8. To add subscription form to your homepage go to **Design->Layouts->Home** and add `Smaily for Opencart` to your prefered position.
 9. That's it, your Opencart store is now integrated with Smaily Plugin!
 
+## Using Cron to Automate Customer Synchronization and Sending Abandoned Carts Emails
+
+To use cron for customer synchronization and for sending abandoned carts emails automatically follow opencart instructions for [setting up cron](http://docs.opencart.com/en-gb/extension/cron/). Url for cron is provided in Smaily settings.
+
+You need to provide a **token** as a parameter for security reasons so that only you can run cronjobs. Cron token is an url-parameter that is checked when cron runs. Enter your token in Smaily admin page under **Subscriber Synchronization->Customer Cron Token** and **Abandoned Cart->Abandoned Cart Token** fields. Token can be any random text/number combination.
+
+We recoment to run Customer Synchronization daily and Abandoned Carts not more often than every hour.
```
Comment (kaittodesk) [inline readme.md]: recommend

File: opencart 3.0.0.0 - 3.0.3.1/upload/admin/controller/extension/module/smaily_for_opencart.php (hunk 1 above)
Comment (sinukaarel) [reply opencart 3.0.0.0 - 3.0.3.1/upload/admin/controller/extension/module/smaily_for_opencart.php]: Moved subdomain escaping after cleaning https://github.com/sendsmaily/smaily-opencart-module/pull/1/commits/5bedbd12ccd5810dcc735616738eab07290ce006

File: readme.md (hunk 2 above)
Comment (sinukaarel) [reply readme.md]: Fixed https://github.com/sendsmaily/smaily-opencart-module/pull/1/commits/5bedbd12ccd5810dcc735616738eab07290ce006

**Pull request merged**